presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java

```diff
@@ -775,6 +795,64 @@ private void queryCompletedEvent(Optional<ExecutionFailureInfo> failureInfo)
             queryInfoOutputPath.ifPresent(path -> writeQueryInfo(path, queryInfo, queryInfoJsonCodec));
         }
 
+        private void processShuffleStats()
+        {
+            List<PrestoSparkShuffleStats> statsList = new ArrayList<>(shuffleStatsCollector.value());
+            Map<ShuffleStatsKey, List<PrestoSparkShuffleStats>> statsMap = new HashMap<>();
+            Set<Integer> fragments = new HashSet<>();
+            for (PrestoSparkShuffleStats stats : statsList) {
+                ShuffleStatsKey key = new ShuffleStatsKey(stats.getFragmentId(), stats.getOperation());
+                statsMap.computeIfAbsent(key, (ignored) -> new ArrayList<>()).add(stats);
+                fragments.add(stats.getFragmentId());
+            }
+            List<Integer> sortedFragments = new ArrayList<>(fragments);
+            sortedFragments.sort(null);
+            log.info("Shuffle statistics summary:");
+            for (Integer fragment : sortedFragments) {
+                List<PrestoSparkShuffleStats> readStatistics = statsMap.get(new ShuffleStatsKey(fragment, READ));
+                if (readStatistics != null) {
+                    logShuffleStatsSummary(fragment, READ, readStatistics);
+                }
+                List<PrestoSparkShuffleStats> writeStatistics = statsMap.get(new ShuffleStatsKey(fragment, WRITE));
+                if (writeStatistics != null) {
+                    logShuffleStatsSummary(fragment, WRITE, writeStatistics);
+                }
+            }
+        }
+
+        private void logShuffleStatsSummary(int fragmentId, Operation operation, List<PrestoSparkShuffleStats> statsList)
+        {
+            long totalProcessedRows = 0;
+            long totalProcessedBytes = 0;
+            long totalElapsedWallTimeMills = 0;
+            for (PrestoSparkShuffleStats stats : statsList) {
+                totalProcessedRows += stats.getProcessedRows();
+                totalProcessedBytes += stats.getProcessedBytes();
+                totalElapsedWallTimeMills += stats.getElapsedWallTimeMills();
+            }
+            long totalElapsedWallTimeSeconds = totalElapsedWallTimeMills / 1000;
+            long rowsPerSecond = totalProcessedRows;
+            long bytesPerSecond = totalProcessedBytes;
+            if (totalElapsedWallTimeSeconds > 0) {
+                rowsPerSecond = totalProcessedRows / totalElapsedWallTimeSeconds;
+                bytesPerSecond = totalProcessedBytes / totalElapsedWallTimeSeconds;
+            }
+            long averageRowSize = 0;
+            if (totalProcessedRows > 0) {
+                averageRowSize = totalProcessedBytes / totalProcessedRows;
+            }
+            log.info(
+                    "Fragment: %s, Operation: %s, Rows: %s, Size: %s, Avg Row Size: %s, Time: %s, %srows/s, %s/s",
```

Comment:
%s rows/s, %s bytes/s

Reply:
This is same as `60.55kB/s`. We print it without a space. Same for rows/s -> `2000rows/s`